pulsar-client/src/main/java/org/apache/pulsar/client/impl/ConsumerImpl.java

```diff
@@ -1643,6 +1643,14 @@ void receiveIndividualMessagesFromBatch(BrokerEntryMetadata brokerEntryMetadata,
                         singleMessageMetadata, uncompressedPayload, batchMessage, schema, true,
                         ackBitSet, ackSetInMessageId, redeliveryCount, consumerEpoch);
                 if (message == null) {
+                    if (ackBitSet != null && !ackBitSet.get(i)) {
```

Comment:
We'd better add a new method, such as `boolean isSingleMessageAcked(BitSetRecyclable ackBitSe)`

So that we can use the new method here and `newSingleMessage()` method.

Reply:
Then we can simplify here to

if (!isSingleMessageAcked()) {
        skippedMessages++;
}
continue;

Reply:
Fixed, Thanks.